internal/app/osd/internal/reg/reg.go

```diff
@@ -321,40 +322,46 @@ func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err
 
 // Routes implements the proto.OSDServer interface.
 func (r *Registrator) Routes(ctx context.Context, in *empty.Empty) (data *proto.RoutesReply, err error) {
-	routeList, err := netlink.RouteList(nil, 2)
+	conn, err := rtnetlink.Dial(nil)
 	if err != nil {
-		return nil, err
+		return nil, errors.Errorf("failed to open socket to rtnetlink: %v", err)
 	}
+	defer conn.Close()
 
-	routes := []*proto.Route{}
+	list, err := conn.Route.List()
+	if err != nil {
+		return nil, errors.Errorf("failed to get route list: %v", err)
+	}
 
-	for _, route := range routeList {
-		link, _err := netlink.LinkByIndex(route.LinkIndex)
-		if _err != nil {
-			err = _err
-			return nil, err
-		}
+	var routes []*proto.Route
 
-		destination := "0.0.0.0"
-		if route.Dst != nil {
-			destination = route.Dst.String()
-		}
+	for _, rMesg := range list {
 
-		gateway := "0.0.0.0"
-		if route.Gw != nil {
-			gateway = route.Gw.String()
+		var ifaceName string
+		ifaceData, err := conn.Link.Get(rMesg.Attributes.OutIface)
+		if err != nil {
+			return nil, errors.Errorf("failed to get interface details for interface index %d: %v", rMesg.Attributes.OutIface, err)
 		}
-
-		routeMessage := &proto.Route{
-			Interface:   link.Attrs().Name,
-			Destination: destination,
-			Gateway:     gateway,
+		if ifaceData.Attributes != nil {
+			ifaceName = ifaceData.Attributes.Name
 		}
-		routes = append(routes, routeMessage)
-	}
 
-	data = &proto.RoutesReply{Routes: routes}
```

Comment:
we do have a named return in the func definition, so we probably want to make use of it or drop it.

Reply:
Quite right, thanks; given the previous return branches, the cleanest is just to remove the return definitions.  If this is a style problem, though, I'm happy to revert.